apps/src/templates/lessonOverview/activities/ActivitySection.jsx

```diff
@@ -42,6 +43,18 @@ export default class ActivitySection extends Component {
 
     const sectionHasTips = section.tips.length > 0;
 
+    let tipsTotalLength = 0;
+    section.tips.forEach(tip => {
+      tipsTotalLength += tip.markdown.length;
+    });
+    const totalLengthOfSectionText = section.text.length + tipsTotalLength;
+    // The width of the tip based on the length of the text of the tip and the activity section
+    // The minimum width the activity section can have is 25
+    const tipWidth = Math.min(
+      Math.round((tipsTotalLength / totalLengthOfSectionText) * 100),
+      75
+    );
```

Comment:
Cool! I am interested to see how well this heuristic performs. If this heuristic is not good enough, there are also some places where we render the element and then measure its height, like here: https://github.com/code-dot-org/code-dot-org/blob/ff60929539ab99cf89f5a97748bc33f6c521cdc3/apps/src/templates/feedback/LevelFeedbackEntry.jsx#L87-L92

If you go that route, you have to be careful not to get into an infinite loop, because changing the width can change the height. There are ways to deal with this, but we can probably put off solving that for now.